benchmarking/process_json_to_csv.jl

```
@@ -50,116 +51,91 @@ import FirstOrderLp
+  :l2_dual_residual,
+  :l_inf_dual_residual,
+  :relative_l2_primal_residual,
+  :relative_l_inf_primal_residual,
+  :relative_l2_dual_residual,
+  :relative_l_inf_dual_residual,
+  :l_inf_primal_variable,
+  :l2_primal_variable,
+  :l_inf_dual_variable,
+]
+
+function solve_log_to_dict(log::FirstOrderLp.SolveLog)::Dict
+  result = Dict()
+  copy_fields_to_dict(result, log, SOLVE_LOG_FIELDS_TO_COPY)
+  result[:cumulative_kkt_matrix_passes] =
     log.solution_stats.cumulative_kkt_matrix_passes
 
   point_type = log.solution_type
   # TODO: This doesn't properly handle the case of infeasibility certificates,
   # whose stats are in log.solution_stats.infeasibility_information.
   for convergence_information in log.solution_stats.convergence_information
     if convergence_information.candidate_type == point_type
-      set_matching_fields(row, convergence_information)
+      copy_fields_to_dict(
+        result,
+        convergence_information,
+        CONVERGENCE_INFORMATION_FIELDS_TO_COPY,
+      )
       break
     end
   end
-  return row
-end
-
-function rows_to_dataframe(rows::Vector{CsvRow})
-  return DataFrame(; StructArrays.components(StructArrays.StructArray(rows))...)
+  return result
 end
 
-struct DatasetNameAndLocation
-  name::String
+struct DatasetConfigAndLocation
+  # Keys must match up with config_labels.
+  config::Dict{String,String}
   logs_directory::String
 end
 
 struct DatasetList
-  datasets::Vector{DatasetNameAndLocation}
+  datasets::Vector{DatasetConfigAndLocation}
+  config_labels::Vector{String}
 end
 
-StructTypes.StructType(::Type{DatasetNameAndLocation}) = StructTypes.Struct()
+StructTypes.StructType(::Type{DatasetConfigAndLocation}) = StructTypes.Struct()
 StructTypes.StructType(::Type{DatasetList}) = StructTypes.Struct()
 
 function read_dataset(dataset_list::DatasetList)::DataFrame
-  rows = CsvRow[]
+  rows = Dict[]
   for dataset in dataset_list.datasets
     logs_directory = dataset.logs_directory
+    experiment_label =
```

Would it help users understand their mistake to have an explicit test that the keys of dataset.config == dataset_list.config_labels and corresponding error message?  This would provide a better diagnostic when c isn't a key in dataset.config, and avoid silently ignoring extra keys within dataset.config.

---

Added an assertion here, which will at least point a user in the right direction if the input is invalid.